### PR TITLE
Add support preload fonts

### PR DESCRIPTION
**Pull request by hasan-ahani**
Hi Spatie team,

Firstly, thank you for developing and maintaining this package!
I've added support for preload functionality to this package. This enhancement enables the preloading of fonts.
Looking forward to your feedback and suggestions.

Best regards

### CODE REVIEW

File: src/GoogleFonts.php [hunk 1]
```diff
@@ -84,21 +94,26 @@ protected function fetch(string $url, ?string $nonce): Fonts
...
             $localizedCss = str_replace(
                 $fontUrl,
-                $this->filesystem->url($this->path($url, $localizedFontUrl)),
+                $url,
                 $localizedCss,
             );
         }
 
         $this->filesystem->put($this->path($url, 'fonts.css'), $localizedCss);
+        $this->filesystem->put($this->path($url, 'preload.txt'), $preloadMeta);
```
Comment (sebastiandedeyne) [inline src/GoogleFonts.php:107]: Could this be an HTML file to clarify the contents?

File: src/Fonts.php [hunk 2]
```diff
@@ -9,11 +9,13 @@
...
-        protected string  $googleFontsUrl,
-        protected ?string $localizedUrl = null,
-        protected ?string $localizedCss = null,
-        protected ?string $nonce = null,
-        protected bool    $preferInline = false,
+        protected string    $googleFontsUrl,
+        protected ?string   $localizedUrl = null,
+        protected ?string   $localizedCss = null,
+        protected ?string   $nonce = null,
+        protected bool      $preferInline = false,
```
Comment (sebastiandedeyne) [inline src/Fonts.php:16]: Can you avoid the whitespace changes?

**Review by sebastiandedeyne — CHANGES_REQUESTED** [review]

File: src/GoogleFonts.php (hunk 1 above)
Comment (hasan-ahani) [reply src/GoogleFonts.php]: Yes, changing the file extension to .html would clarify the contents and make it more understandable, especially if the file contains HTML markup or content that is better represented in an HTML format.

File: src/Fonts.php (hunk 2 above)
Comment (hasan-ahani) [reply src/Fonts.php]: Absolutely, I've made sure to avoid any whitespace changes in the commit.

**Review by sebastiandedeyne — APPROVED** [review]

**Comment by hasan-ahani** [comment]
I've addressed the test errors, and everything should be working fine now. Please feel free to let me know if you encounter any further issues. Thanks😊
@freekmurze

**Comment by sebastiandedeyne** [comment]
Thanks!

**Pull request merged**